src/api.rs

```diff
@@ -1,5 +1,66 @@
+pub use std::convert::TryFrom;
 use std::error::Error;
 
+bitflags! {
+    pub struct Access: u32 {
+        const R = 0b00000001;
+        const W = 0b00000010;
+        const X = 0b00000100;
+        const NIL = 0b00000000;
+        const DUMMY = 0b11111111;
+        const RW = Self::R.bits | Self::W.bits;
+        const WX = Self::W.bits | Self::X.bits;
+        const RX = Self::R.bits | Self::X.bits;
+        const RWX = Self::R.bits | Self::W.bits | Self::X.bits;
+    }
+}
+
+impl TryFrom<Access> for u8 {
+    type Error = &'static str;
+    fn try_from(access: Access) -> Result<Self, Self::Error> {
+        let p_access: u8;
+        p_access = match access {
+            Access::NIL => 0,
+            Access::R => 1,
+            Access::W => 2,
+            Access::RW => 3,
+            Access::X => 4,
+            Access::RX => 5,
+            Access::WX => 6,
+            Access::RWX => 7,
+            _ => 8,
+        };
+        if p_access == 8 {
+            Err("not valid access!")
+        } else {
+            Ok(p_access)
+        }
+    }
+}
+
+impl TryFrom<u8> for Access {
+    type Error = &'static str;
+    fn try_from(access: u8) -> Result<Self, Self::Error> {
+        let p_access: Access;
+        p_access = match access {
+            0 => Access::NIL,
+            1 => Access::R,
+            2 => Access::W,
+            3 => Access::RW,
+            4 => Access::X,
+            5 => Access::RX,
+            6 => Access::WX,
+            7 => Access::RWX,
+            _ => Access::DUMMY,
+        };
+        if p_access == Access::DUMMY {
+            Err("not valid access!")
+        } else {
+            Ok(p_access)
+        }
```

Comment:
It looks to me like you have introduced the `DUMMY` value for the sole purpose of being able to match on the fallthrough case. This is not good style. Why not simply:
```suggestion
        match access {
            0 => Ok(Access::NIL),
            1 => Ok(Access::R),
            2 => Ok(Access::W),
            3 => Ok(Access::RW),
            4 => Ok(Access::X),
            5 => Ok(Access::RX),
            6 => Ok(Access::WX),
            7 => Ok(Access::RWX),
            _ => Err("invalid access value"),
        }
```

Reply:
Also you should not use plain integers if there are `KVMI_PAGE_ACCESS` definitions available in the kvmi crate.